src/crewai/agent.py

```diff
@@ -264,15 +267,63 @@ def execute_task(
                     task=task,
                 ),
             )
-            result = self.agent_executor.invoke(
-                {
-                    "input": task_prompt,
-                    "tool_names": self.agent_executor.tools_names,
-                    "tools": self.agent_executor.tools_description,
-                    "ask_for_human_input": task.human_input,
-                }
-            )["output"]
+
+            if not isinstance(self.max_execution_time, int) or self.max_execution_time <= 0:
+                raise ValueError("Max Execution time must be a positive integer greater than zero")
+            
+            # Apply timeout if specified
+            if self.max_execution_time is not None:
+                import concurrent.futures
+                with concurrent.futures.ThreadPoolExecutor() as executor:
+                    future = executor.submit(
+                        self.agent_executor.invoke,
+                        {
+                            "input": task_prompt,
+                            "tool_names": self.agent_executor.tools_names,
+                            "tools": self.agent_executor.tools_description,
+                            "ask_for_human_input": task.human_input,
+                        }
+                    )
+                    
+                    try:
+                        result = future.result(timeout=self.max_execution_time)["output"]
+                    except concurrent.futures.TimeoutError:
+                        # Cancel the execution if possible
+                        future.cancel()
+                        crewai_event_bus.emit(
+                            self,
+                            event=AgentExecutionErrorEvent(
+                                agent=self,
+                                task=task,
+                                error=f"Task '{task.description}' execution timed out after {self.max_execution_time} seconds. Consider increasing max_execution_time or optimizing the task.",
+                            ),
+                        )
+                        raise TimeoutError(f"Task '{task.description}' execution timed out after {self.max_execution_time} seconds. Consider increasing max_execution_time or optimizing the task.")
+                    except Exception as error:
+                        future.cancel()
+                        crewai_event_bus.emit(
+                            self,
+                            event=AgentExecutionErrorEvent(
+                                agent=self,
+                                task=task,
+                                error=f"Task execution failed: {str(error)}"
```

Comment:
I'm thinking if we need this block. Because we are handling exception at line +323. 
Assuming that this is not an TimeoutError, we can try drop this except block. wdyt?